_articles/scripting/Typescript/typescript-modifier.md

```diff
@@ -0,0 +1,231 @@
+    "AbilityManaCost"				"80 90 100 110"
+
+    // Special
+    //-------------------------------------------------------------------------------------------------------------
+    "AbilitySpecial"
+    {
+        "01"
+        {
+          "var_type"				"FIELD_INTEGER"
+          "resist_debuff"			"30 35 40 45"
+        }
+
+        "02"
+        {
+          "var_type"				"FIELD_FLOAT"
+          "seal_duration"			"3.0 4.0 5.0 6.0"
+        }
+    }
+    "AbilityCastAnimation"		"ACT_DOTA_CAST_ABILITY_3"
+}
+```
+
+As the `ScriptFile` denotes it, the lua file should be in `vscripts/abilities/`. To do so, we'll create our TS file in `src/vscripts/abilities/`, where it would be appropriately routed when compiled. Create the `typescript_skywrath_mage_ancient_seal.ts` file and open it.
+
+
+### Coding The Ability
+
+The ability itself is very straightforward, since all it does is apply a modifier on the target. For simplicity sake, let's decide the modifier will be named `modifier_typescript_ancient_seal`. Following is the ability:
+
+```ts
+import { BaseAbility, registerAbility } from "../lib/dota_ts_adapter";
+
+@registerAbility()
+export class typescript_skywrath_mage_ancient_seal extends BaseAbility {
+	sound_cast = "Hero_SkywrathMage.AncientSeal.Target";
+
+	OnSpellStart() {
+		// Special values
+		const seal_duration = this.GetSpecialValueFor("seal_duration");
+
+		// Fetch target
+		const target = this.GetCursorTarget()!;
+
+		// Play sound
+		target.EmitSound(this.sound_cast);
+
+		// Add modifier
+		target.AddNewModifier(this.GetCaster(), this, "modifier_typescript_ancient_seal", { duration: seal_duration });
+	}
+}
+```
+
+Great! This applies the modifier on the target. But we haven't defined the modifier yet, so let's do that next.
+
+### Creating The Modifier
+
+This part is absolutely up to you and your organizational preferences: some prefer to add the modifier as a separate file, while some prefer to have the ability and its associated modifiers in the same file. You could place the modifier file inside `src/vscripts/modifiers`, for instance. In order to keep the guide simple, let's make the modifier in the same file.
+
+Very similar to an ability in TS, modifiers are also a class. We create a modifier with the following structure:
+
+```ts
+@registerModifier()
+export class modifier_typescript_ancient_seal extends BaseModifier {
+
+}
+```
+
+As you can see, it's very similar to an ability, replacing `@registerAbility()` with `@registerModifier()`, and the `BaseAbility` extension with `BaseModifier`.
+Note that `@registerModifier()` takes care of LinkLuaModifier for you, so you don't need to call it on TS modifiers.
+
+### Typechecking Modifier Calls
+
+Before we continue, one thing we can do to add some more type checking is to link the ability to the modifier. Instead of relying on a string for the naming of the modifier, we'll instead link the class name.
```

Comment:
Actually this is kinda required, if the modifier is not required anywhere it will not be linked and stuff will fail. You should probably just state that this is the absolute truth and should not be skipped.

Reply:
Changed. It is now phrased as something that must be done.